service/src/main/java/bio/terra/catalog/datarepo/DatarepoClient.java

```diff
@@ -0,0 +1,38 @@
+package bio.terra.catalog.datarepo;
+
+import bio.terra.catalog.config.DatarepoConfiguration;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.datarepo.api.SnapshotsApi;
+import bio.terra.datarepo.api.UnauthenticatedApi;
+import bio.terra.datarepo.client.ApiClient;
+import javax.ws.rs.client.Client;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DatarepoClient {
+  private final DatarepoConfiguration datarepoConfig;
+  private final Client commonHttpClient = new ApiClient().getHttpClient();
+
+  public DatarepoClient(DatarepoConfiguration datarepoConfig) {
+    this.datarepoConfig = datarepoConfig;
+  }
+
+  private ApiClient getApiClient(AuthenticatedUserRequest user) {
+    ApiClient apiClient = getApiClient();
+    apiClient.setAccessToken(user.getToken());
+    return apiClient;
+  }
+
+  private ApiClient getApiClient() {
+    // Share one api client across requests.
+    return new ApiClient().setHttpClient(commonHttpClient).setBasePath(datarepoConfig.basePath());
+  }
+
```

Comment:
why get rid of the @visiblefortesting tag?

Reply:
It's only needed if the access is changed due to the need to access a member from test code. We used to need it to make some APIs visible that we would've made private, because there wasn't a way to mock the point where an API was created. Now that this is done in a separate class, the access would no longer be private so the annotation isn't needed.